### ISSUE TITLE
Forgot password route crashes

### ISSUE THREAD

**Issue by hristokoev** [payloadcms/payload#8185]
### Environment Info

```text
Payload: beta.104
Node.js: 22.8.0
Next.js: 15.0.0-canary.104
```

### Describe the Bug

When trying to access the `/admin/forgot` route, I get the following error:

```
Unhandled Runtime Error
Error: Functions cannot be passed directly to Client Components unless you explicitly expose it by marking it with "use server". Or maybe you meant to call this function rather than return it.
  <... elements={{...}} i18nKey=... t={function t}>
                                      ^^^^^^^^^^^^
```

Full error in the console:

```
 ⨯ Error: Functions cannot be passed directly to Client Components unless you explicitly expose it by marking it with "use server". Or maybe you meant to call this function rather than return it.
  <... elements={{...}} i18nKey=... t={function t}>
                                      ^^^^^^^^^^^^
    at stringify (<anonymous>)
    at stringify (<anonymous>)
    at stringify (<anonymous>)
    at AsyncResource.runInAsyncScope (node:async_hooks:211:14)
digest: "3967774931"
 GET /admin/forgot 500 in 254ms

```

### Reproduction Steps

Simply install the latest beta version (beta.104) and go to `/admin/forgot`.

I believe this started around beta.98, ~~because I have another project with beta.97 and the route there works~~ It works only in production, but in dev it's still there.

**Comment by paulpopus** [comment]
I couldn't reproduce this on the latest beta. Do you have any custom components by any chance that could affect this page?

**Comment by hristokoev** [comment]
I've used the following command:

`npx create-payload-app@beta my-project`

Then I selected the blank template and installed. Upon running, I had to create a user, then I logged in and tried the route `/admin/forgot`, which resulted in the same crash, though it showed a different error on the frontend (in terminal the error is still the same as written above). There are **no customisations** whatsoever.

![image](https://github.com/user-attachments/assets/4b54e128-5ec5-4a56-b92a-2c1964c06eff)
